File: src/main/java/pl/training/workshop/m6/s04_encapsulatefactory/step1/tickets/Tickets.java
package pl.training.workshop.m6.s04_encapsulatefactory.step1.tickets;

import pl.training.workshop.shared.Money;

/** Krok 1: metody tworzące (Creation Method) obok klas biletów - zwracają typ Ticket. */
public final class Tickets {
    private Tickets() {
    }

    public static Ticket standard(String title, Money base, int row) {
        return new StandardTicket(title, base, row);
    }

    public static Ticket vip(String title, Money base, int row) {
        return new VipTicket(title, base, row);
    }
}

File: csharp/src/Training.Workshop/M6/S04EncapsulateFactory/Step1/Ticketing/Tickets.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S04EncapsulateFactory.Step1.Ticketing;

/// <summary>Krok 1: metody tworzące (Creation Method) obok klas biletów - zwracają typ ITicket.</summary>
public static class Tickets
{
    public static ITicket Standard(string title, Money basePrice, int row)
    {
        return new StandardTicket(title, basePrice, row);
    }

    public static ITicket Vip(string title, Money basePrice, int row)
    {
        return new VipTicket(title, basePrice, row);
    }
}
